web-api/src/business/useCases/caseInventoryReport/getCustomCaseReportInteractor.test.ts

```diff
@@ -1,14 +1,16 @@
+jest.mock('@web-api/persistence/elasticsearch/getCasesByFilters');
 import {
   GetCustomCaseReportRequest,
   getCustomCaseReportInteractor,
 } from './getCustomCaseReportInteractor';
-import { applicationContext } from '../../../../../shared/src/business/test/createTestApplicationContext';
+import { getCasesByFilters as getCasesByFiltersMock } from '@web-api/persistence/elasticsearch/getCasesByFilters';
```

Comment:
Just a nit, but I think you have the naming reversed, at least, in how I would think of it ... this import is the _real_ `getCasesByFilters` - as it is importing from the implementation module, right? Below, the return of `jest.mocked` is _not_ `getCasesByFilters`, rather it is, presumably, the mock of it.

Reply:
This works a little differently from how you think, but maybe there is a better way to get that across?

as soon as you do jest.mock('@web-api/persistence/elasticsearch/getCasesByFilters') then wherever you import '@web-api/persistence/elasticsearch/getCasesByFilters' you are getting a mock. So the import you pointed out IS actually a mock, but Typescript does not know that it thinks you have imported the real thing. jest.mocked() Doesn't actually do anything but tell typescript "hey buddy this function is actually mocked".